hamamatsu.py

```diff
@@ -552,21 +553,21 @@ def data_out(self) -> str:
         hm = "Hamamatsu"
         hm_str = ""
         sz = self.last_measurement.shape
-        hm_str += format_data(f"{hm}/numShots", f"{sz[0]}")
-        hm_str += format_data(f"{hm}/rows", f"{sz[1]}")
-        hm_str += format_data(f"{hm}/columns", f"{sz[2]}")
+        hm_str += TCP.format_data(f"{hm}/numShots", f"{sz[0]}")
+        hm_str += TCP.format_data(f"{hm}/rows", f"{sz[1]}")
+        hm_str += TCP.format_data(f"{hm}/columns", f"{sz[2]}")
 
         for shot in range(sz[0]):
             flat_ar = np.reshape(self.last_measurement[shot, :, :], sz[1]*sz[2])
-            tmp_str = ar_to_str(flat_ar)  # TODO : implement based on encoding in cspy and labview
-            hm_str += format_data(f"{hm}/shots/{shot}", tmp_str)
+            tmp_str = u16_ar_to_str(flat_ar)
```

Comment:
Further up it's not assumed the array will be u16 (u8 and u32 are options) but for the Hamamatsu I'm pretty sure the data is in u16. 
If you can make the u8 and u32 methods I can set up the logic checking so the conversion goes smoothly in the general case.

Reply:
Was the logic that auto-forced images to u16 not preserved?

Reply:
I should note that I chose this because CsPy assumes u16. CsPy should fundamentally fail if it received another data type.

Reply:
Oh jeez. Ok I'll add a little error message if the data format isn't u16 in that case

Reply:
An error message isn't necessary. Simply casting the array to u16 once the data is taken during the measurement will suffice.

Reply:
And that is the strategy currently implemented by LabVIEW

Reply:
Ok, taking a look, the data array is currently filled with python ints so the casting should be the same in all cases

Reply:
Does your u16 function perform the casting or should that be done explicitly?